src/main/java/org/broadleafcommerce/payment/service/gateway/DefaultPayPalCheckoutConfiguration.java

```diff
@@ -1,107 +1,91 @@
-/*
- * #%L
- * BroadleafCommerce PayPal
- * %%
- * Copyright (C) 2009 - 2014 Broadleaf Commerce
- * %%
- * Licensed under the Broadleaf Fair Use License Agreement, Version 1.0
- * (the "Fair Use License" located  at http://license.broadleafcommerce.org/fair_use_license-1.0.txt)
- * unless the restrictions on use therein are violated and require payment to Broadleaf in which case
- * the Broadleaf End User License Agreement (EULA), Version 1.1
- * (the "Commercial License" located at http://license.broadleafcommerce.org/commercial_license-1.1.txt)
- * shall apply.
- * 
- * Alternatively, the Commercial License may be replaced with a mutually agreed upon license (the "Custom License")
- * between you and Broadleaf Commerce. You may not use this file except in compliance with the applicable license.
- * #L%
- */
 package org.broadleafcommerce.payment.service.gateway;
 
-import org.broadleafcommerce.common.config.service.SystemPropertiesService;
-import org.broadleafcommerce.common.payment.PaymentGatewayType;
-import org.broadleafcommerce.common.payment.service.AbstractPaymentGatewayConfiguration;
-import org.broadleafcommerce.common.web.BaseUrlResolver;
 import org.broadleafcommerce.vendor.paypal.service.payment.MessageConstants;
 import org.broadleafcommerce.vendor.paypal.service.payment.PayPalCheckoutPaymentGatewayType;
 import org.broadleafcommerce.vendor.paypal.service.payment.type.PayPalShippingDisplayType;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Service;
+import org.springframework.core.env.Environment;
+
+import com.broadleafcommerce.paymentgateway.domain.enums.PaymentGatewayType;
+import com.broadleafcommerce.paymentgateway.service.configuration.AbstractPaymentGatewayConfiguration;
 
 import java.net.URI;
 import java.net.URISyntaxException;
 import java.util.HashMap;
 import java.util.Map;
 
-import javax.annotation.Resource;
+import lombok.RequiredArgsConstructor;
 
 /**
  * @author Elbert Bautista (elbertbautista)
  */
-@Service("blPayPalCheckoutConfiguration")
-public class PayPalCheckoutConfigurationImpl extends AbstractPaymentGatewayConfiguration implements PayPalCheckoutConfiguration {
-
-    @Resource(name = "blBaseUrlResolver")
-    protected BaseUrlResolver urlResolver;
+@RequiredArgsConstructor
+public class DefaultPayPalCheckoutConfiguration extends AbstractPaymentGatewayConfiguration
+        implements PayPalCheckoutConfiguration {
 
-    @Autowired
-    protected SystemPropertiesService propertiesService;
+    private final Environment env;
```

Comment:
I'm struggling with what we should do here. On the one hand, it would be nice to maintain the ability to merge changes up into this branch from the legacy branch, in which case we should leave the code similar to how it was before. On the other hand, the way we do properties has changed a lot with our microservices patterns, and specifically how those properties are documented. For example, what I want to say is that you should refactor _all_ of the properties in this entire module to use `@ConfigurationProperties` instead and add Javadocs to each of those properties indicating what they do and how they are used instead of relying on the `Environment`.

@elbertbautista what do you think we should do here?

Reply:
Either route has a bad side unfortunately. I lean towards refactoring to use `@ConfigurationProperties` so that the available configuration is clearly communicated to customers, but I'm not thrilled about the potential for tricky merging situations. 

At the same time, this isn't the only item that would lead to difficult merging. That leads me to:
- What other changes would cause a merge to be challenging? - updated package structures, refactoring of the `PaymentGatewayServiceProvider`, etc.
- How much more of a challenge would this change add?

Reply:
>  I lean towards refactoring to use @ConfigurationProperties so that the available configuration is clearly communicated to customers, but I'm not thrilled about the potential for tricky merging situations.

Perhaps we should just treat this as it's _not_ a merge at all. Instead, the previous version will continue to exist exactly as it has in perpetuity, but in maintenance mode. This version of the library will evolve and change where _it_ makes sense.

Coming back to this I think this is the right answer and continued mergability is not really that much of a concern. @elbertbautista?

Reply:
That sounds right to me.

... leaving this conversation unresolved to give @elbertbautista an opportunity to object.